// Copyright Robert Bosch GmbH, 2021. Part of the Eclipse Kuksa Project.
//
// All rights reserved. This configuration file is provided to you under the
// terms and conditions of the Eclipse Distribution License v1.0 which
// accompanies this distribution, and is available at
// http://www.eclipse.org/org/documents/edl-v10.php


syntax = "proto3";

package kuksa;

// The connecting service definition.
service viss_client {
  rpc GetMetaData (commandRequest) returns (data) {}
  rpc AuthorizeChannel (commandRequest) returns (data) {}
  rpc GetValue (commandRequest) returns (data) {}
  rpc SetValue (commandRequest) returns (data) {}
}

message data {
  string value_ = 1;
  kuksaChannel channel_ = 2;
}

message commandRequest {
  string path_ = 1;
  string reqID_ = 2;
  optional int32 value_ = 3;
  kuksaChannel channel_ = 4;
}

message kuksaChannel{
  enum Type {
    UNKNOWN_TYPE = 0;
    WEBSOCKET_PLAIN = 1;
    WEBSOCKET_SSL = 2;
    HTTP_PLAIN = 3;
    HTTP_SSL = 4;
    GRPC = 5;
  };
  uint64 connectionID = 1;
  bool authorized = 2;
  bool modifyTree = 3;
  bytes authToken = 4;
  string permissions = 5;
  Type typeOfConnection = 6;
}

